### DISCUSSION TITLE
Usage as a library

### DISCUSSION THREAD

**Discussion by tanguc** [GitoxideLabs/gitoxide#921] [Q&A]
### Summary 💡

Hello,

Just tested the tool and it's blazzing fast, awesome thank you guys !

Wondering if the project could be easily used as a library from a Rust project ? is there any docs / examples ?

Thank you

**Comment by Byron** [comment]
The [`gix` docs](https://docs.rs/gix/0.48.0/gix/) serve as entry-point, but from there it's probably still a bit bumpy unless you look at example programs and how `gix::Repository` is used within the codebase itself.

It's definitely a good idea to start with requirements first and see if all features you need are present, or to go from `git2` instead and try to find API counterparts. These could even be contributed to [this list](https://docs.rs/gix/0.48.0/gix/#libgit2-api-to-gix) which right now is far from exhaustive.

I hope that helps.
